### DISCUSSION TITLE
change controls mtr icon

### DISCUSSION THREAD

**Discussion by maoweifan** [fabricjs/fabric.js#7898] [Q&A] [answered]
## Version
5.2.1

<img width="1365" alt="image" src="https://user-images.githubusercontent.com/51221758/164588439-b4ebb4aa-d49d-4c4b-bb65-d1ccd7a214ff.png">

i just want to change mtr icon
but i do this but it doesn't work

what can I do?
I don't want to overwrite the original controls
Just need to change the icon

**Comment by ShaMan123** [comment]
Add a working reproduction. Always.

**Comment by rolitter** [comment]
https://github.com/pixolith/fabricjs-customise-controls-extension
![image](https://user-images.githubusercontent.com/27326998/164676622-353abeb4-8282-4feb-b446-907ca7367dc7.png)

**Comment by asturur** [comment]
@maoweifan if you are using fabricJS 4+ that library isn't compatible anymore.

**Reply by maoweifan** [reply]
not support @5.2?

**Reply by asturur** [reply]
no that i k now of.

**Comment by ShaMan123** [comment, marked as answer]
Take a look:
http://fabricjs.com/custom-control-render
http://fabricjs.com/demos/

**Reply by ShaMan123** [reply]
```js
 fabric.Object.prototype.controls.mtr.render = renderIcon;
 fabric.Object.prototype.controls.mtr.cornerSize = 50;


  function renderIcon(ctx, left, top, styleOverride, fabricObject) {
    var size = this.cornerSize;
    ctx.save();
    ctx.translate(left, top);
    ctx.rotate(fabric.util.degreesToRadians(fabricObject.angle));
    ctx.drawImage(this.icon, -size/2, -size/2, size, size);
    ctx.restore();
  }

```

**Reply by rolitter** [reply]
[Fabricjs Custom Controls](https://codepen.io/rolitter/pen/WNdVKVj)
![image](https://user-images.githubusercontent.com/27326998/165685557-2e0552b9-c480-4b3b-a306-6873bf6f34c4.png)

**Reply by ShaMan123** [reply]
beautiful stuff
